Packs/Zoom/Integrations/Zoom/Zoom.py

```diff
@@ -313,17 +378,400 @@ def zoom_list_user_messages(self, user_id: str, date_arg: datetime, from_arg: da
+        investigation_id = mirror['investigation_id']
+        await handle_text_received_from_zoom(investigation_id, message, operator_email, operator_name)
+
+
+async def event_url_validation(payload):
+    """ Verify the authenticity of a token
+    Args:
+        payload : request from Zoom
+    Returns:
+        json_res: A dictionary containing the 'plainToken' and its corresponding 'encryptedToken' (HMAC-SHA256 signature).
+    """
+    plaintoken = payload.get('plainToken')
+    hash_object = hmac.new(SECRET_TOKEN.encode('utf-8'), msg=plaintoken.encode('utf-8'), digestmod=hashlib.sha256)
+    expected_signature = hash_object.hexdigest()
+    json_res = {
+        "plainToken": plaintoken,
+        "encryptedToken": expected_signature
+    }
+    return json_res
+
+
+@app.post('/')
+async def handle_zoom_response(request: Request, credentials: HTTPBasicCredentials = Depends(basic_auth),
+                               token: APIKey = Depends(token_auth)):
+    """handle any response that came from Zoom app
+    Args:
+        request : zoom request
+    Returns:
+        JSONResponse:response to zoom
+    """
+    request = await request.json()
+    demisto.debug(request)
+    credentials_param = demisto.params().get('credentials')
+    auth_failed = False
+    v_token = demisto.params().get('verification_token', {}).get('password')
+    if 'Basic' not in str(token) and v_token:
+        demisto.debug(v_token)
+        if token != v_token:
+            auth_failed = True
+
+    elif credentials and credentials_param and (username := credentials_param.get('identifier')):
+        password = credentials_param.get('password', '')
+        demisto.debug(username, password)
+        if not compare_digest(credentials.username, username) or not compare_digest(credentials.password, password):
+            auth_failed = True
+    if auth_failed:
+        demisto.debug('Authorization failed ')
+        return Response(status_code=status.HTTP_401_UNAUTHORIZED, content='Authorization failed.')
+
+    event_type = request['event']
+    payload = request['payload']
+    try:
+        if event_type == 'endpoint.url_validation' and SECRET_TOKEN:
+            res = await event_url_validation(payload)
+            return JSONResponse(content=res)
+
+        if event_type == 'interactive_message_actions':
+            if 'actionItem' in payload:
+                action = payload['actionItem']['value']
+            elif 'selectedItems' in payload:
+                action = payload['selectedItems'][0]['value']
+            else:
+                return Response(status_code=status.HTTP_400_BAD_REQUEST)
+            message_id = payload['messageId']
+            account_id = payload['accountId']
+            robot_jid = payload['robotJid']
+            to_jid = payload['toJid']
+            user_name = payload['userName']
+            entitlement_reply = await check_and_handle_entitlement(action, message_id, user_name)
```

Comment:
It's possible that `action` is `None` here.
Is that ok?

Reply:
there is no possibility for action to be None, 
```
if 'actionItem' in payload:
                action = payload['actionItem']['value']
            elif 'selectedItems' in payload:
                action = payload['selectedItems'][0]['value']
            else:
                return Response(status_code=status.HTTP_400_BAD_REQUEST)
```